packages/utils/src/getFeatureFlag.ts

```diff
@@ -0,0 +1,33 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { Analytics } from '@deriv-com/analytics';
+
+const waitForGrowthbook = () => {
+    return new Promise(resolve => {
+        const checkAnalytics = () => {
+            if (typeof Analytics !== 'undefined' && Analytics.getInstances()?.ab !== undefined) {
+                resolve(Analytics);
+            } else {
+                setTimeout(checkAnalytics, 100); // Retry every 100ms
+            }
+        };
+
+        checkAnalytics();
+    });
+};
+
+const getFeatureFlag = async (feature: string, defaultValue?: string | boolean | undefined) => {
+    let enabled = false;
+    await waitForGrowthbook();
+
```

Comment:
I am confused, can you please help me understand the implementation, because `waitForGrowthbook` is where we are checking instance of gb from Analytics. So that is the place where Growthbook with call init() method. And that is a place where error can occur if GB service is down or for any reason, right ?. So in that case our promise will not get resolved ?

Reply:
And we are checking for growthbook status after we are trying to initialise the GB. Will it make it to the line `Analytics?.getGrowthbookStatus();` ?

Reply:
<img width="888" alt="Screenshot 2024-10-25 at 8 25 00 AM" src="https://github.com/user-attachments/assets/4969167f-0e2b-4324-9c94-fbd089d1cf84">

GB.init() method returns these three code , should we not somehow manage to get this status all the way upto the consuming app(deriv-app) rather than us trying to do it manually in analytics pacakge

Reply:
<img width="648" alt="Screenshot 2024-10-25 at 8 27 39 AM" src="https://github.com/user-attachments/assets/3d3b7527-9cfb-4853-aa38-41fcfd5d3273">
 This is what I can see in Analytics package, we are not using the code provided by them.

Reply:
Please help me understand , I might be missing something